tests/test_mirrors.py

```diff
@@ -60,30 +60,36 @@ def setUp(self):
                  'metadata_path' : 'metadata',
                  'targets_path' : 'targets',
                  'confined_target_dirs' : ['targets/release/',
-                                            'targets/release/']}}
+                                            'targets/release/']},
+     # Make sure we are testing if confined_target_dirs is ['']
+     'mirror4': {'url_prefix' : 'http://mirror4.com',
+                 'metadata_path' : 'metadata',
+                 'targets_path' : 'targets',
+                 'confined_target_dirs' : ['']}}
 
 
 
   def test_get_list_of_mirrors(self):
     # Test: Normal case.
     mirror_list = mirrors.get_list_of_mirrors('meta', 'release.txt', self.mirrors)
-    self.assertEqual(len(mirror_list), 3)
+    self.assertEqual(len(mirror_list), 4)
     for mirror, mirror_info in six.iteritems(self.mirrors):
       url = mirror_info['url_prefix'] + '/metadata/release.txt'
       self.assertTrue(url in mirror_list)
 
     mirror_list = mirrors.get_list_of_mirrors('target', 'a.txt', self.mirrors)
-    self.assertEqual(len(mirror_list), 1)
+    self.assertEqual(len(mirror_list), 2)
     self.assertTrue(self.mirrors['mirror1']['url_prefix']+'/targets/a.txt' in \
                     mirror_list)
 
     mirror_list = mirrors.get_list_of_mirrors('target', 'a/b', self.mirrors)
-    self.assertEqual(len(mirror_list), 1)
+    self.assertEqual(len(mirror_list), 2)
     self.assertTrue(self.mirrors['mirror1']['url_prefix']+'/targets/a/b' in \
                     mirror_list)
 
     mirror1 = self.mirrors['mirror1']
     del self.mirrors['mirror1']
+    del self.mirrors['mirror4']
     mirror_list = mirrors.get_list_of_mirrors('target', 'a/b', self.mirrors)
     self.assertFalse(mirror_list)
     self.mirrors['mirror1'] = mirror1
```

Comment:
Do we need to re-add mirror4 here also?

Reply:
We could in the name of consistency, but we can say we don't need to re-add mirror1 too because the following tests are regarding exception handling and don't actually check the content of `self.mirrors`.